### DISCUSSION TITLE
@itwin/web-viewer-react Viewer Error

### DISCUSSION THREAD

**Discussion by joeyflora** [iTwin/community#55] [Q&A] [answered]
I'm seeing an error from inside the `BaseViewer` component when using the @itwin/web-viewer-react library when using version `3.1.5`. 

![Screen Shot 2024-10-17 at 12 53 13 PM](https://github.com/user-attachments/assets/5e5a2e63-a4fe-45c2-9dd9-25b36503f717)

Stepping through the source code it appears to be originating from the call to `getChangeSetFromNamedVersion()` inside the `getVersion()` function within `@itwin/viewer-react/src/services/iModel/IModelService.ts` 

![Screen Shot 2024-10-17 at 12 52 34 PM](https://github.com/user-attachments/assets/2eb73038-bb62-4a96-b5b2-aa07aaa8acbe)

I have confirmed that the iModelId, iTwinId, and authClient props provided to the Viewer component are all valid. Has anyone seen an issue like this before or have ideas on how to troubleshoot? 

Thanks

**Comment by ben-polinsky** [comment]
Hi,

Do you have the actual error ("the above error")?

**Reply by joeyflora** [reply]
Ah yea sorry about that. 

![Screen Shot 2024-10-17 at 2 52 25 PM](https://github.com/user-attachments/assets/04c7ce6a-c116-4499-836c-2f236205fb13)

**Reply by ben-polinsky** [reply]
Nothing comes to mind immediately.

Is there a network call that fails? If so, check the request/response headers in chrome devtools for an x-correlation-id. I could then try to find the request in our logs and see if there is more information. 

Have you been using this app successfully in the past? Does a different iTwin/iModel throw the same error?

Have you attempted to load the same iModel/iTwin/clientId in our create-react-app template viewer? It's a bit easier to setup than a custom integration. 

https://github.com/iTwin/viewer/tree/master/packages/templates/cra-template-web-viewer
Getting started: https://www.itwinjs.org/learning/tutorials/develop-web-viewer/

**Reply by joeyflora** [reply]
Looks like there are two request that are returning 404s. One to the imodels/xxx/changesets with x-correlation-id 0111a3c1-ef0f-4ff7-b921-350707151509 and one to imodels/xxx/namedversions with x-correlation-id ec19c90f-46f1-4ff3-90fb-93f8efa1afc5.  I built this app maybe a year and a half ago with the create-react-app template but it hasn't been used in a while and I've been updating some deprecated functionality. I'm able to view the same iTwinid/iModelid in the web view at https://connect-imodelweb.bentley.com/. Let me know if you have any other suggestions.

**Reply by ben-polinsky** [reply]
Seems like you're hitting our old v1 `/iModels` endpoint. Your requests do produce errors about a subscription being expired, but since you can access the same iModel through the web I'm guessing it's the older API that is triggering the error. 

If you're providing a `hubAccess` prop to the `Viewer`, add the version `itwin-platform.v2`. If not providing that prop, try adding it:

```tsx
const hubAccess = new FrontendIModelsAccess(
  new IModelsClient({
  api: {
    baseUrl: `https://api.bentley.com/imodels`,
    version: "itwin-platform.v2"
  },
})
      
<Viewer 
  hubAccess={hubAccess}
... 
/>
```

**Reply by aruniverse** [reply, marked as answer]
Another option is to update the version of your dependency on "@itwin/imodels-access-frontend" and "@itwin/imodels-client-management" to the latest
